src/lib/containers/FilePreview/EntityPreview.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { useResolveLinkEntity } from '../../utils/functions/ResolveLinkEntity'
+import useGetEntityBundle from '../../utils/hooks/SynapseAPI/entity/useEntityBundle'
+import { ALL_ENTITY_BUNDLE_FIELDS } from '../../utils/SynapseConstants'
+import { FileEntity, isFileEntity } from '../../utils/synapseTypes'
+import { SynapseSpinner } from '../LoadingScreen'
+import FileEntityPreview from './FileEntityPreview'
+
+export type EntityPreviewProps = {
+  entityId: string
+  versionNumber?: number
+}
+
+/**
+ * Returns a preview of an entity. Currently only works for FileEntities, and Links that resolve to FileEntities.
+ * @param props
+ * @returns
+ */
+export default function EntityPreview(props: EntityPreviewProps) {
```

Comment:
Just takes a synID and ensures it's a file or a link that points to a file

Reply:
I see how this works for a Link that points to a FileEntity, but I don't see how this works for an ID that points to a FileEntity.  Could you explain?

Reply:
Oh, I see.  It looks like resolveLinkEntity() will get the FileEntity and just return it